### ISSUE TITLE
UseDataClass triggered on class with private constructor and static initializers

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#791]
I'm getting `UseDataClass` on a class that has a private constructor and static initializers wrapped with `@JvmStatic` inside an companion object. I'd argue that this should not be a data class and hence not get flagged.

**Comment by schalkms** [comment]
I agree that classes with only private constructors should not be a data class candidate. I am not quite sure about candidate classes with static initializers.

**Comment by schalkms** [comment]
Okay. What do you think about data class candidates with a static initializer? Should detekt also exclude those?

**Comment by vanniktech** [comment]
Only using the private constructor as an exclusion point is fine with me. If you want static factory methods for something you can use a private constructor anyways.

We might even exclude `internal` ones since afaik if you use them you won't get a synthetic method generated.
